src/protocol/context/upgrade.rs

```diff
@@ -44,17 +41,6 @@ use std::{
 /// let _ = <UpgradeContext<C<'_, F>, F, NoRecord> as UpgradeToMalicious<Vec<Replicated<F>>, _>>::upgrade;
 /// let _ = <UpgradeContext<C<'_, F>, F, NoRecord> as UpgradeToMalicious<(Vec<Replicated<F>>, Vec<Replicated<F>>), _>>::upgrade;
 /// ```
-///
-/// ```compile_fail
-/// use ipa::protocol::{context::{UpgradeContext, UpgradeToMalicious, UpgradedMaliciousContext as C}, NoRecord, RecordId};
-/// use ipa::ff::Fp32BitPrime as F;
-/// use ipa::secret_sharing::replicated::{
-///     malicious::AdditiveShare as MaliciousReplicated, semi_honest::AdditiveShare as Replicated,
-/// };
-/// // This can't be upgraded with a record-bound context because the record ID
-/// // is used internally for vector indexing.
-/// let _ = <UpgradeContext<C<'_, F>, F, RecordId> as UpgradeToMalicious<Vec<Replicated<F>>, _>>::upgrade;
```

Comment:
I couldn't work out how to make this keep failing.  The way it failed before was that we only implemented for `Vec<Vec<T>>`, but now that there is a generic `IntoIterator` version, I'm not sure what we would do to keep this.

Reply:
Using a non-`Vec` type that is `IntoIterator` (like `[Replicated<F>; 3]`) would work, but I'm not sure how much value that has. I might be inclined to delete all of these doc tests -- without the negative case, I don't think the positive cases have much value.

Reply:
Test restoration incoming.  I forgot to revert this part of the change.